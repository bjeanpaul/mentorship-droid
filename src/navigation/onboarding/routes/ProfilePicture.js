import { connect } from 'react-redux';
import ProfilePicture from 'src/profile/ProfilePicture';
import * as navigate from '../actions';

const mapStateToProps = state => ({
  imagePath: state.profile.newProfileImagePath,
});

export default connect(mapStateToProps, {
  onChoosePhotoPress: navigate.pushCameraRollRoute,
})(ProfilePicture);
