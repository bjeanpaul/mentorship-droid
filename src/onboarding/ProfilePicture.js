import React, {
  Component,
} from 'react';

import { Link } from 'react-router-native';

import {
  View,
  Image,
  TouchableHighlight,
} from 'react-native';

import {
  Text,
  Heading,
  Button,
} from 'src/components';


class ProfilePicture extends Component {

  constructor(props) {
    super(props);
    this.state = {
      profileSource: this.props.location.query.uri || "app/assets/Profile_Add.png",
    };
  }

  render() {

    console.log(this.props.location.query.uri)

    return (
      <View style={{ flex: 1 }}>
        <Heading>Add a profile picture</Heading>
        <Image
          style={{
            width: 100,
            height: 100,
            margin: 10,
          }}
          source={{ uri: this.state.profileSource }}
        />
        <Link to="camera-roll">
          <Button>Take Photo</Button>
        </Link>
      </View>
    );
  }
}

export default ProfilePicture;
