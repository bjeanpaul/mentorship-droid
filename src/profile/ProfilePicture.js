import React from 'react';
import { View, Image, StyleSheet } from 'react-native';

import { BaseView, Toolbar, Text } from 'src/components';


const styles = StyleSheet.create({
  imageContainer: {
    alignItems: 'center',
  },
  image: {
    width: 120,
    height: 120,
    marginBottom: 19,
  },
  imageLabel: {
    color: '#9fb1b3',
    fontSize: 18,
  },
});

// TODO: Picture path should be dynamic.
const ProfilePicture = ({
}) => {
  const IMG_PROFILE_ADD = require('app/assets/Profile_Add.png');
  return (
    <BaseView>
      <Toolbar title="Add a profile picture" />
      <View style={styles.imageContainer}>
        <Image
          style={styles.image}
          source={IMG_PROFILE_ADD}
        />
        <Text style={styles.imageLabel}>Choose Photo</Text>
      </View>
    </BaseView>
  );
};
ProfilePicture.propTypes = {
  picturePath: React.PropTypes.string,
};

export default ProfilePicture;
