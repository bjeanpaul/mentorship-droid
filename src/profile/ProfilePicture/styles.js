import { StyleSheet } from 'react-native';
import colors from 'src/constants/colors';

export default StyleSheet.create({
  imageContainer: {
    alignItems: 'center',
  },
  image: {
    width: 120,
    height: 120,
    marginBottom: 19,
  },
  imageLabel: {
    color: colors.PROFILE_PICTURE_CHOOSE_PHOTO_TEXT,
    fontSize: 18,
  },
});
