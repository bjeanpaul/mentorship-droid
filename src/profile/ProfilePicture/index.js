import React from 'react';
import { View, Image, TouchableWithoutFeedback } from 'react-native';

import { BaseView, Toolbar, Text, Pagination } from 'src/components';
import images from 'src/constants/images';
import styles from './styles';


const ProfilePicture = ({
  imagePath,
  onChoosePhotoPress,
}) => (
  <BaseView>
    <Toolbar title="Add a profile picture" />
    <TouchableWithoutFeedback onPress={onChoosePhotoPress}>
      <View style={styles.imageContainer}>
        <Image
          style={styles.image}
          source={
            imagePath && { uri: imagePath } ||
            images.PROFILE_FACE_ADD
          }
        />
        <Text style={styles.imageLabel}>Choose Photo</Text>
        </View>
      </TouchableWithoutFeedback>
      <Pagination
        onBackPress={() => console.log('12312312')}
        onNextPress={() => console.log('12312312')}
      />
  </BaseView>
);
ProfilePicture.propTypes = {
  imagePath: React.PropTypes.string,
  onChoosePhotoPress: React.PropTypes.func.isRequired,
};

export default ProfilePicture;
