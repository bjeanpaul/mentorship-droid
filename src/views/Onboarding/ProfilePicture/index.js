import React, { PropTypes } from 'react';
import { View, Image, TouchableWithoutFeedback } from 'react-native';

import { Text, FormStep } from 'src/components';
import images from 'src/constants/images';
import styles from './styles';


const ProfilePicture = ({
  profile: { profilePic },
  onChoosePhotoPress,
  ...props,
}) => (
  <FormStep
    title="Add a profile picture"
    paginationDisabled={!profilePic}
    {...props}
  >
    <TouchableWithoutFeedback onPress={onChoosePhotoPress}>
      <View style={styles.imageContainer}>
        <Image
          style={styles.image}
          source={
            profilePic
              ? { uri: profilePic }
              : images.PROFILE_FACE_ADD
          }
        />
        <Text style={styles.imageLabel}>Choose Photo</Text>
      </View>
    </TouchableWithoutFeedback>
  </FormStep>
);


ProfilePicture.propTypes = {
  profile: PropTypes.object.isRequired,
  onChoosePhotoPress: PropTypes.func.isRequired,
  onBackPress: PropTypes.func.isRequired,
  onNextPress: PropTypes.func.isRequired,
};


export default ProfilePicture;
