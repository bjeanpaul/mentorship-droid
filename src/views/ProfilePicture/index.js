import React, { PropTypes } from 'react';
import { View, Image, TouchableWithoutFeedback } from 'react-native';

import { BaseView, Header, Text, Pagination, ProgressBar } from 'src/components';
import images from 'src/constants/images';
import styles from './styles';


const ProfilePicture = ({
  imagePath,
  onBackPress,
  onNextPress,
  onChoosePhotoPress,
}) => (
  <BaseView>
    <ProgressBar completed={1 / 7} />
    <Header>
      <Text style={Text.types.title}>Add a profile picture</Text>
    </Header>

    <TouchableWithoutFeedback onPress={onChoosePhotoPress}>
      <View style={styles.imageContainer}>
        <Image
          style={styles.image}
          source={
            imagePath && { uri: imagePath } ||
            images.PROFILE_FACE_ADD
          }
        />
        <Text style={styles.imageLabel}>Choose Photo</Text>
      </View>
    </TouchableWithoutFeedback>

    <View style={styles.footer}>
      <Pagination
        onBackPress={onBackPress}
        onNextPress={onNextPress}
        disabled={!imagePath}
      />
    </View>
  </BaseView>
);
ProfilePicture.propTypes = {
  imagePath: React.PropTypes.string,
  onChoosePhotoPress: React.PropTypes.func.isRequired,
  onBackPress: PropTypes.func.isRequired,
  onNextPress: PropTypes.func.isRequired,
};

export default ProfilePicture;
