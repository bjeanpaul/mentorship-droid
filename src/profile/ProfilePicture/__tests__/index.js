import React from 'react';
import { shallow } from 'enzyme';
import ProfilePicture from '../index';

describe('ProfilePicture', () => {
  function noop() {}

  it('should map props correctly', () => {
    expect(
      <ProfilePicture
        imagePath="path/to/image.png"
        onChoosePhotoPress={noop}
      />
    ).toMatchSnapshot();
  });

  it('should be able to tap and fire `onChoosePhotoPress`', () => {
    const onChoosePhotoPress = jest.fn();
    const el = shallow(
      <ProfilePicture
        onChoosePhotoPress={onChoosePhotoPress}
      />
    );
    el.find('TouchableWithoutFeedback').simulate('press');
    expect(onChoosePhotoPress).toBeCalled();
  });
});
