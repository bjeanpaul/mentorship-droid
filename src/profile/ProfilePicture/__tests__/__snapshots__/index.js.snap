exports[`ProfilePicture should map props correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Object {
        "height": 72,
        "justifyContent": "center",
        "paddingLeft": 16,
        "paddingRight": 16
      }
    }>
    <Image
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      source={1}
      style={undefined}
      testID={undefined} />
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text_medium",
            "fontSize": 20,
            "fontWeight": "400"
          }
        ]
      }>
      Add a profile picture
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "alignItems": "center"
      }
    }
    testID={undefined}>
    <Image
      source={
        Object {
          "uri": "path/to/image.png"
        }
      }
      style={
        Object {
          "height": 120,
          "marginBottom": 19,
          "width": 120
        }
      } />
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#9fb1b3",
            "fontSize": 18
          }
        ]
      }>
      Choose Photo
    </Text>
  </View>
</View>
`;
